internal/controller/grafanaorganization_controller.go

```diff
@@ -180,10 +208,45 @@ func (r GrafanaOrganizationReconciler) reconcileCreate(ctx context.Context, graf
 		logger.Info("updated orgID in the grafanaOrganization status")
 	}
 
-	// Configure the organization's datasources and authorization settings
-	err = grafanaService.SetupOrganization(ctx, grafanaOrganization)
+	// Configure the organization's datasources and handle status updates
+	datasources, err := grafanaService.ConfigureDatasources(ctx, organization)
 	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("failed to setup grafanaOrganization: %w", err)
+		return ctrl.Result{}, fmt.Errorf("failed to configure datasources: %w", err)
+	}
+
+	// Build the list of configured datasources for the status
+	var configuredDatasources = make([]v1alpha1.DataSource, len(datasources))
+	for i, datasource := range datasources {
+		configuredDatasources[i] = v1alpha1.DataSource{
+			ID:   datasource.ID,
+			Name: datasource.Name,
+		}
+	}
+
+	// Sort the datasources by ID to ensure consistent ordering
+	slices.SortStableFunc(configuredDatasources, func(a, b v1alpha1.DataSource) int {
+		return cmp.Compare(a.ID, b.ID)
+	})
+
+	// Update the status if the datasources have changed
+	if !slices.Equal(grafanaOrganization.Status.DataSources, configuredDatasources) {
+		logger.Info("updating datasources in the GrafanaOrganization status")
+		grafanaOrganization.Status.DataSources = configuredDatasources
+		if err := r.Client.Status().Update(ctx, grafanaOrganization); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to update GrafanaOrganization status: %w", err)
+		}
+		logger.Info("updated datasources in the GrafanaOrganization status")
+	}
+
+	// Configure SSO settings for all organizations
+	allOrganizations, err := r.listActiveGrafanaOrganizations(ctx)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to get all organizations for SSO configuration: %w", err)
+	}
```

Comment:
This should be within ConfigureSSOSettings IMO, avoid calling this before hand all the time.